src/app/qgsmaptooladdpart.cpp

```diff
@@ -194,6 +198,12 @@ void QgsMapToolAddPart::cadCanvasReleaseEvent( QgsMapMouseEvent * e )
       vlayer->endEditCommand();
 
       vlayer->triggerRepaint();
+
+      if (( !isGeometryEmpty ) && QgsWkbTypes::isSingleType( vlayer->wkbType() ) )
```

Comment:
`isGeometryEmpty` is used only here. For the sake of clarity and compactiy, better do directly

     if ((vlayer->selectedFeatures()[0].geometry().isEmpty() ) && QgsWkbTypes::isSingleType( vlayer->wkbType() ) )